_posts/2021-05-22-fluxv2-openfaas.md

```diff
@@ -0,0 +1,332 @@
+You wont be able to expose your functions on the internet without some extra software like [inlets](https://inlets.dev) to provide a public IP.
+
+Create a `kind` cluster abd wait for it to be ready.
+
+```sh
+kind create cluster \
+  --name flux-openfaas
+  
+kind export kubeconfig --name flux-openfaas 
+```
+
+# Install Flux
+Flux uses a CLI to setup and install the flux controllers into the cluster. Once we have the CLI we can 
+install the controllers and then start adding resources into our cluster! We installed the `flux1 cli` earlier, so we are ready to go.
+
+We have the Flux cli we need to `bootstrap` the cluster and create a git repository. 
+
+For this section I'm using Github, you can find instructions for other Git providers [in the flux docs](https://fluxcd.io/docs/installation/#bootstrap)
+
+Generate a `github personal access token` with `repo` permissions [like this](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
+
+Export this as an environment variable
+
+```sh 
+export GITHUB_TOKEN=<your-token>
+export GITHUB_USER=<your github username>
+```
+
+Run the bootstrap for a repository on your personal GitHub account:
+
+```sh 
+flux bootstrap github \
+  --owner=$GITHUB_USER \
+  --repository=openfaas-flux \
+  --path=clusters/linode \
+  --personal
+```
+
+After a short time we should see the flux controllers installed into the `flux-system` namespace!
+
+```sh 
+$ kubectl get pods -n flux-system
+NAME                                       READY   STATUS    RESTARTS   AGE
+helm-controller-85bfd4959d-w5ccr           1/1     Running   0          38s
+kustomize-controller-6977b8cdd4-gblq7      1/1     Running   0          38s
+notification-controller-5c4d48f476-7tdkk   1/1     Running   0          38s
+source-controller-85fb864746-gzrxq         1/1     Running   0          38s
+```
+
+This step has:
+* Installed the `flux controllers` into your cluster 
+* Created a github repository, named `openfaas-flux` (unless you changed that). 
+* Added a `Deploy key` to Github with access to that repository
+
+These flux controllers are using the deploy key to read changes in the github repository and install things based on 
+changes! In the next section we will add some `yaml` manifests into the git repository and see Flux install these into the cluster.
+
+# Install OpenFaaS
+
+Clone your git repository, for me the commands were
+
+```sh 
+git clone git@github.com:$GITHUB_USER/openfaas-flux.git
+cd openfaas-flux/clusters/linode
+```
+
+Our flux controllers are syncing manifests we change in the `clusters/linode` directory that we specified as the path on isntall.
+
+We can now add a `HelmRepository` and `HelmRelease` to install the OpenFaaS Chart
+
+You can get more information on the [flux components in the docs](https://fluxcd.io/docs/components/)
+
+Create our OpenFaaS namespaces by adding a file with these contents to the repository called `openfaas-namespaces.yaml`
```

Comment:
Link to the source file please:

https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml

Perhaps have them download it instead of repeating it in the blog post, otherwise we have to maintain it in multiple places.